uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/calendar-model_test.js

```diff
@@ -25,29 +29,49 @@ define([
       $rootScope.$apply();
     });
 
-    describe('getCalendar()', function () {
+    describe('get()', function () {
       var CalendarPromise,
         deferred;
 
-      function commonSetUp(returnData) {
+      function commonSetUp (returnData) {
         deferred = $q.defer();
         deferred.resolve(returnData);
         WorkPatternAPI.getCalendar.and.returnValue(deferred.promise);
 
         CalendarPromise = Calendar.get(jasmine.any(String), jasmine.any(String), jasmine.any(Object));
       }
 
-      it('calls equivalent API method', function () {
-        commonSetUp(mockData.daysData());
-        CalendarPromise.then(function () {
-          expect(WorkPatternAPI.getCalendar).toHaveBeenCalled();
+      describe('basic tests', function () {
+        it('calls equivalent API method', function () {
+          commonSetUp(mockData.daysData());
+          CalendarPromise.then(function () {
+            expect(WorkPatternAPI.getCalendar).toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('when passing a single contact id', function () {
+        it('returns a single CalendarInstance', function () {
+          commonSetUp(mockData.daysData());
+          CalendarPromise.then(function (response) {
+            expect(_.isArray(response)).toBe(false);
+            expect('days' in response).toBe(true);
+          });
         });
       });
 
-      it('returns model instances when request is successful', function () {
-        commonSetUp(mockData.daysData());
-        CalendarPromise.then(function (response) {
-          expect('days' in response).toBe(true);
+      describe('when passing multiple contact ids', function () {
+        beforeEach(function () {
+          WorkPatternAPI.getCalendar.and.returnValue($q.resolve(mockData.daysData()));
+          CalendarPromise = Calendar.get([jasmine.any(String), jasmine.any(String)], jasmine.any(String));
+        });
+
+        it('returns multiple CalendarInstances', function () {
+          CalendarPromise.then(function (response) {
+            expect(_.isArray(response)).toBe(true);
+            expect('days' in response[0]).toBe(true);
```

Comment:
What if we change the no of responses from mocked data. We should loop on `response` here.